### PR TITLE
Chapter 11 - Account activation

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Activate account via the activation link in email for sign up user

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class UsersController < ApplicationController
   before_action :admin_user, only: :destroy
 
   def index
-    @users = User.sort_by_name.page params[:page]
+    @users = User.sort_by_name.page params[:page].per(10)
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: move 10 to settings.yml file and call here

File: app/models/user.rb [hunk 2]
```diff
@@ -41,14 +41,28 @@ def forget
     update_attribute :remember_digest, nil
   end
 
-  def authenticated? remember_token
-    return false unless remember_digest
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false unless digest
+    BCrypt::Password.new(digest).is_password? token
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
    return false unless digest
    
    BCrypt::Password.new(digest).is_password? token
```

**Comment by EvonneNgXinYi** [comment]
ready

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class UserMailer < ApplicationMailer
+
```
Comment (saobv-2223) [inline app/mailers/user_mailer.rb]:
```suggestion
```

**Review by saobv-2223 — COMMENTED** [review]
fix the small syntax and merge

**Pull request merged**